cmd/root.go

```diff
@@ -65,16 +78,31 @@ func init() {
 
 func initConfig() {
 	if cfgFile != "" {
+		viper.SetConfigType("yaml")
+		emptyConfigBytes, err := yaml.Marshal(config)
+		if err != nil {
+			LogError("Can't marshal config:", err.Error())
+			os.Exit(1)
+		}
+
+		emptyConfigReader := bytes.NewReader(emptyConfigBytes)
+		viper.MergeConfig(emptyConfigReader)
```

Comment:
If I'm reading that viper bug correctly, the `MergeConfig` call needs to be done even if `cfgFile == ""`.

Reply:
I've tried already few times as well, I just switched to defining them manually: https://github.com/SUSE/eirini-loggregator-bridge/pull/14/commits/32059c959d6ec84588329bf5f3927c8effd94326

Reply:
See https://github.com/spf13/viper/issues/761 to check how UnMarshal in Viper works under the hood